server.go

```diff
@@ -1953,6 +1966,45 @@ func (p dataNodes) Len() int           { return len(p) }
 func (p dataNodes) Less(i, j int) bool { return p[i].ID < p[j].ID }
 func (p dataNodes) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
 
+// Authorize user u to execute query q on database.
+// database can be "" for queries that do not require a database.
+func Authorize(u *User, q *influxql.Query, database string) error {
+	// Cluster admins can do anything.
```

Comment:
By struggling, I meant trying to communicate clearly to next dev that a `nil` user meant no auth was actually enabled.